<refactor> Move test case scenarios into provider

roleyfoley opened the pull request:
Another move around for getting the testing idea working 

Moves the loading processing of scenarios into the ability to load additional providers as SubProviders. SubProviders load like normal providers except when it comes to output management which is set by default to the existing GENERATION_PROVIDER variable 

Along with working for scenarios, this allows for codeontap users to load their own providers on top of standard providers, overriding default values or implementing their own components while still using the resources of another provider 

From a testing side this introduces a new provider `awstest` which contains scenarios designed to test the templates that we generate in a unit testing style, while keeping the testing isolated from the real world provider code

ml019 commented:
@roleyfoley Went to merge this but looks like it needs tweaking based on the discussion on providers yesterday?

roleyfoley commented:
removed subProviders ( GENERATION_SUBPROVIDERS)  and moved from `GENERATION_PROVIDER` to `GENERATION_PROVIDERS` which takes a comma separated list of providers 

The first provider in the list is used when determining outputs etc

Had to update the genplan output to go off the command line list of providers instead of determining them based on the placement

ml019 commented on aws/createBlueprint.sh:
```diff
@@ -30,17 +30,15 @@ DESCRIPTION:
     -h                         shows this text
 (o) -o OUTPUT_DIR              is the directory where the outputs will be saved - defaults to the PRODUCT_STATE_DIR
-(o) -p GENERATION_PROVIDER     is the provider to for template generation
+(o) -p GENERATION_PROVIDERS     is the provider to for template generation
```
Normally we just allow the switch multiple times and gather the values e.g. -v switches for freemarker.sh

roleyfoley replied:
Fixed up now accepts a list of -p values or an env var `GENERATION_PROVIDERS`

roleyfoley commented:
@ml019 ready when you are on this one?
